src/jit/optimizer.cpp

```diff
@@ -1416,410 +1418,1018 @@ void Compiler::optCheckPreds()
 
 #endif // DEBUG
 
-/*****************************************************************************
- * Find the natural loops, using dominators. Note that the test for
- * a loop is slightly different from the standard one, because we have
- * not done a depth first reordering of the basic blocks.
- */
-
-void Compiler::optFindNaturalLoops()
+namespace
 {
-#ifdef DEBUG
-    if (verbose)
+//------------------------------------------------------------------------
+// LoopSearch: Class that handles scanning a range of blocks to detect a loop,
+//             moving blocks to make the loop body contiguous, and recording
+//             the loop.
+//
+// We will use the following terminology:
+//   HEAD    - the basic block that flows into the loop ENTRY block (Currently MUST be lexically before entry).
+//             Not part of the looping of the loop.
+//   FIRST   - the lexically first basic block (in bbNext order) within this loop.  (May be part of a nested loop,
+//             but not the outer loop. ???)
```

Comment:
I don't understand the question.  Is the question whether `fist` may be part of "a nested loop", and also whether it may be part of "the outer loop"?  If so, "nested" and "outer" with respect to what?  Why is it "**a** nested loop" vs. "**the** outer loop"?  If we're talking about the full loop nest forest, then there aren't any restrictions on how deeply nested a loop with a "first" is.  If we're talking about whether a given loop's "first" may be included in any "outer" loops enclosing the given one, then yes of course it is a member of all of them.  If we're talking about whether a given loop's "first" may be included in any "nested" loops enclosed by the given one, then:
 - as far as `LoopSearch` is concerned, there's nothing conceptually barring that
 - there's code called after `loopSearch`, but still within `optFindNaturalLoops`, that canonicalizes loops; its comments say the point is to have no two loops share `top` but then the implementation talks about sharing `fist`
 - in the current implementation, `first` and `top` are always the same, and the search algorithm in `LoopSearch` will never find two loops that share one

So I don't know what was supposed to be said here.  Should I just remove the whole parenthetical?